pkg/agent/api/grpc/hyperstart.proto

```diff
@@ -0,0 +1,135 @@
+syntax = "proto3";
+
+package grpc;
+
+import "oci.proto";
+import "google/protobuf/empty.proto";
+
+// To generate hyperstart.pb.go, run:
+// $ go get -u github.com/google/protobuf
+// $ protoc -I=$GOPATH/src/github.com/google/protobuf/src/ -I. --go_out=plugins=grpc:. hyperstart.proto oci.proto
+
+// unstable
+service HyperstartService {
+	// execution
+	rpc CreateContainer(CreateContainerRequest) returns (google.protobuf.Empty);
+	rpc StartContainer(StartContainerRequest) returns (google.protobuf.Empty);
+	rpc ExecProcess(ExecProcessRequest) returns (google.protobuf.Empty);
+	rpc SignalProcess(SignalProcessRequest) returns (google.protobuf.Empty);
+	rpc WaitProcess(WaitProcessRequest) returns (WaitProcessResponse); // wait & reap like waitpid(2)
+
+	// stdio
+	rpc WriteStdin(WriteStreamRequest) returns (WriteStreamResponse);
+	rpc ReadStdout(ReadStreamRequest) returns (ReadStreamResponse);
+	rpc ReadStderr(ReadStreamRequest) returns (ReadStreamResponse);
+	rpc CloseStdin(CloseStdinRequest) returns (google.protobuf.Empty);
+	rpc TtyWinResize(TtyWinResizeRequest) returns (google.protobuf.Empty);
+
+	// misc (TODO: some rpcs can be replaced by hyperstart-exec)
+	rpc StartSandbox(StartSandboxRequest) returns (google.protobuf.Empty);
+	rpc DestroySandbox(DestroySandboxRequest) returns (google.protobuf.Empty);
+	rpc UpdateInterface(UpdateInterfaceRequest) returns (google.protobuf.Empty);
+	rpc AddRoute(AddRouteRequest) returns (google.protobuf.Empty);
+	rpc OnlineCPUMem(OnlineCPUMemRequest) returns (google.protobuf.Empty);
+}
+
+message CreateContainerRequest {
+	string container_id = 1;
+	Spec OCI = 2;
+}
+
+message StartContainerRequest {
+	string container_id = 1;
+}
+
+message ExecProcessRequest {
+	string container_id = 1;
+	Process process = 2;
+}
+
+message SignalProcessRequest {
+	string container_id = 1;
+	string process_id = 2;
+	uint32 signal = 3;
+}
+
+message WaitProcessRequest {
+	string container_id = 1;
+	string process_id = 2;
+}
+
+message WaitProcessResponse {
+	int32 status = 1;
+}
+
+message WriteStreamRequest {
+	string container_id = 1;
+	string process_id = 2;
+	bytes data = 3;
+}
+
+message WriteStreamResponse {
+	uint32 len = 1;
+}
+
+message ReadStreamRequest {
+	string container_id = 1;
+	string process_id = 2;
+	uint32 len = 3;
+}
+
+message ReadStreamResponse {
+	bytes data = 1;
+}
+
+message CloseStdinRequest {
+	string container_id = 1;
+	string process_id = 2;
+}
+
+message TtyWinResizeRequest {
+	string container_id = 1;
+	string process_id = 2;
+	uint32 row = 3;
+	uint32 column = 4;
+}
+
+message StartSandboxRequest {
+	string hostname = 1;
+	repeated string dns = 2;
```

Comment:
`+ repeated Storage`

Reply:
@laijs Yes, we need that one.

Reply:
why we need it in both Create and Start API?

Reply:
We need it for the `StartSandbox` for storage endpoints that are shared across all containers and in `CreateContainer` for container specific ones.

Reply:
hmm...

should we change `StartSandbox` to `CreateSandbox`? It looks a bit strange to start without a create. @sameo @laijs

Reply:
Here `StartSandbox` is actually a create and start operation. But I'm fine if we rename it `CreateSandbox()`. It's also consistent with `DestroySandbox()`.

Reply:
I think the "Sandbox" has some runtime semantics and it is not persisted, which could be created as running, paused, or destroyed, and there should not be a created and stopped sandbox in our context.

Reply:
@gnawux Right. So what I'm saying is that I'm fine if we only have:
```
	// misc (TODO: some rpcs can be replaced by hyperstart-exec)
	rpc CreateSandbox(CreateSandboxRequest) returns (google.protobuf.Empty);
	rpc DestroySandbox(DestroySandboxRequest) returns (google.protobuf.Empty);
```

because in our case we're only going to create and destroy sandboxes.

Are you suggesting we should have:

```
	rpc CreateSandbox(CreateSandboxRequest) returns (google.protobuf.Empty);
	rpc StartSandbox(StartSandboxRequest) returns (google.protobuf.Empty);
	rpc StopSandbox(StopSandboxRequest) returns (google.protobuf.Empty);
	rpc DestroySandbox(DestroySandboxRequest) returns (google.protobuf.Empty);
```

?